### PR TITLE
Openmp

### PR DESCRIPTION
**Pull request by TApplencourt**

### CODE REVIEW

File: omp/tracer_omp.sh.in [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+display_version() {
+    cat $datadir/version
+    exit 1
+}
+
+while true; do
+  case "$1" in
+    --help           ) display_help; exit;;
```
Comment (Kerilk) [inline omp/tracer_omp.sh.in:24]: Missing option to trace Intel specific callbacks.

File: omp/tracer_omp.sh.in [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  lttng stop
+  lttng destroy
+  exit
+}
+
+trap ctrl_c INT
+
+if [ ! -z "$lttng_view" ]
+then
+  lttng view --viewer="babeltrace_cuda --context --live" &
```
Comment (Kerilk) [inline omp/tracer_omp.sh.in:61]: Wrong viewer.

File: utils/xprof_utils.hpp [hunk 3]
```diff
@@ -5,6 +5,24 @@
...
+enum backend_e{ BACKEND_UNKNOW = 0,
+                BACKEND_ZE = 1,
+                BACKEND_OPENCL = 2,
+                BACKEND_CUDA = 3,
+                BACKEND_OMP_TRANSFER = 4,
+                BACKEND_OMP = 5 };
+
+constexpr int backend_level[] = { 2, 2, 2, 2, 1, 0 };
+
+constexpr const char* backend_name[] = { "BACKEND_UNKNOW",
```
Comment (Kerilk) [inline utils/xprof_utils.hpp:17]:
```suggestion
constexpr const char* backend_name[] = { "BACKEND_UNKNOWN",
```

File: utils/xprof_utils.hpp [hunk 4]
```diff
@@ -89,9 +107,9 @@ thread_id_t borrow_thread_id(const bt_event*);
 
 bt_message* create_host_message(const char *hostname, const process_id_t, const thread_id_t,
                                 const char *name, const uint64_t ts, const uint64_t duration, const bool err,
-                                bt_event_class*, bt_self_message_iterator*, bt_stream*);
+                                bt_event_class*, bt_self_message_iterator*, bt_stream*, backend_t = BACKEND_UNKNOW);
```
Comment (Kerilk) [inline utils/xprof_utils.hpp:110]:
```suggestion
                                bt_event_class*, bt_self_message_iterator*, bt_stream*, backend_t = BACKEND_UNKNOWN);
```

File: utils/xprof_utils.hpp [hunk 5]
```diff
@@ -101,7 +119,7 @@ bt_message* create_device_name_message(const char* hostname, const process_id_t
 
 bt_message* create_traffic_message(const char *hostname, const process_id_t, const thread_id_t,
                                    const char *name, const uint64_t size,
-                                   bt_event_class*, bt_self_message_iterator*, bt_stream*);
+                                   bt_event_class*, bt_self_message_iterator*, bt_stream*, backend_t = BACKEND_UNKNOW);
```
Comment (Kerilk) [inline utils/xprof_utils.hpp:122]:
```suggestion
                                   bt_event_class*, bt_self_message_iterator*, bt_stream*, backend_t = BACKEND_UNKNOWN);
```

File: utils/xprof_utils.hpp [hunk 6]
```diff
@@ -5,6 +5,24 @@
 #include <string>
 #include "babeltrace2/babeltrace.h"
 
+enum backend_e{ BACKEND_UNKNOW = 0,
```
Comment (Kerilk) [inline utils/xprof_utils.hpp:8]:
```suggestion
enum backend_e{ BACKEND_UNKNOWN = 0,
```

File: xprof/timeline.cpp [hunk 7]
```diff
@@ -78,31 +84,81 @@ bt_component_class_sink_consume_method_status timeline_dispatch_consume(
...
+                                                     {"tid", gtf_tid},
+                                                     { "args", { {"sort_index", level } } },
+                                                     // Double "{" to generate a dict and not array
+                                                   }
+                                  << "," << std::endl;
+                    } else {
+                        gtf_tid = it->second;
+                    }
+                }
+*/
```
Comment (Kerilk) [inline xprof/timeline.cpp:156]: Any reason to keep this?

**Review by Kerilk — CHANGES_REQUESTED** [review]

File: omp/tracer_omp.sh.in (hunk 1 above)
Comment (TApplencourt) [reply omp/tracer_omp.sh.in:30]: Fixed by #f595184

File: omp/tracer_omp.sh.in [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+prefix=@prefix@
+datarootdir=@datarootdir@
+datadir=@datadir@
+
+display_help() {
+    echo "Usage:"
+    echo " $(basename $0) [options] [--] <application> <application-arguments>"
+    echo "  --help                        Show this screen"
+    echo "  --version                     Print the version string"
+    exit 1
```
Comment (TApplencourt) [inline omp/tracer_omp.sh.in:14]:
```suggestion
    echo "  --disable-intel-extensions          disable-intel-extensions "
    exit 1
```

Comment (Kerilk) [inline omp/tracer_omp.sh.in:14]:
```suggestion
    echo "  --help                        Show this screen"
    echo "  --version                     Print the version string"
    echo "  --disable-intel-extensions    Disable Intel extensions"
    exit 1
```

File: omp/tracer_omp.sh.in (hunk 1 above)
Comment (Kerilk) [reply omp/tracer_omp.sh.in]: Fixed by https://github.com/argonne-lcf/THAPI/pull/33/commits/cfbef8db8a5c9abf4831447f4f56de886d70542c

**Pull request merged**